services/user.py

```diff
@@ -0,0 +1,36 @@
+from django.contrib.auth import get_user_model
+from django.contrib.auth.models import User
+
+def create_user(username: str, password: str, email: str = None, first_name: str = None, last_name: str = None) -> User:
+    user = get_user_model().objects.create_user(username=username, password=password)
+
+    if email:
+        user.email = email
+    if first_name:
+        user.first_name = first_name
+    if last_name:
+        user.last_name = last_name
+
+    user.set_password(password)

```

Comment:
You're calling `set_password()` on the user object, but this is unnecessary because `create_user()` already sets the password. You can remove this line.